src/Omnipay/Common/Message/AbstractRequest.php

```diff
@@ -271,21 +281,67 @@ public function setCardReference($value)
     }
 
     /**
-     * Get the payment amount.
+     * Convert an amount into a float.
      *
-     * @return string
+     * @var string|int|float $value The value to convert.
+     * @throws InvalidRequestException on any validation failure.
+     * @return float The amount converted to a float.
+     */
+
+    public function toFloat($value)
+    {
+        if (!is_string($value) && ! is_int($value) && ! is_float($value)) {
+            throw new InvalidRequestException('Data type is not a valid decimal number.');
+        }
+
+        if (is_string($value)) {
+            // Validate generic number, with optional sign and decimals.
+            if (!preg_match('/^[-]?[0-9]+(\.[0-9]*)?$/', $value)) {
+                throw new InvalidRequestException('String is not a valid decimal number.');
+            }
+        }
+
+        return (float)$value;
+    }
```

Comment:
The `InvalidRequestException` seems like the wrong exception to raise on a validation error in the Helper class. `ValidationException` seems more appropriate. Possibly `\InvalidArgumentException` in the first instance (if the supplied type is not a int/string/float).

Can this Helper method call those exceptions directly, or do ALL exceptions raised explicitly by OmniPay logic need to be extended into the OmniPay namespace?